osquery/database/plugins/rocksdb.cpp

```diff
@@ -114,6 +120,41 @@ class RocksDBDatabasePlugin : public DatabasePlugin {
 /// Backing-storage provider for osquery internal/core.
 REGISTER_INTERNAL(RocksDBDatabasePlugin, "database", "rocksdb");
 
+class RocksDBGuard {
+ public:
+  RocksDBGuard() : my_tid_(platformGetTid()), locked_(false) {}
+
+  bool safeEnter() {
+    if (!owner_tid_is_valid_ || owner_tid_ != my_tid_) {
+      mu_.lock();
```

Comment:
I should ask - please advice on a path forward. My original submission simply used Google Logging's RAW_LOG_INFO, the Google recommended way of ensuring thread-safe logging. The downside here, of course, is that all RocksDB logs would just go to stderr.

The current PR captures the RocksDB logs, but sacrifices concurrent rocksdb operations. Given that osquery calls into rocksdb at a per-log-line level, and the relatively large-grained concurrency architecture of osquery itself (eg., all scheduled queries run in a single thread), the current PR's concurrency constrain does not seem to be particularly restrictive.